metaflow/plugins/events_decorator.py

```diff
@@ -188,6 +241,63 @@ def flow_init(
 
         # TODO: Handle scenario for local testing using --trigger.
 
+    def format_deploytime_value(self):
+        for trigger in self.triggers:
+            # Case were trigger is a function that returns a list of events
+            # Need to do this bc we need to iterate over list later
+            if isinstance(trigger, DeployTimeField):
+                if isinstance(deploy_time_eval(trigger), list):
+                    deploy_val = deploy_time_eval(trigger)
+                    self.triggers.remove(trigger)
+                    self.triggers.extend(deploy_val)
+            else:
+                break
+        for trigger in self.triggers:
+            old_trigger = trigger
+            # Entire event is a function (returns either string or dict)
+            if isinstance(trigger, DeployTimeField):
```

Comment:
This should not happen since we removed the DeployTimeField in the first step. Now we need to go inside each dict (we either have a fully resolved dict or a dict with some portions that are not yet evaluated).